app/services/proxy_deployment_service.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+class ProxyDeploymentService
+  delegate :apicast_configuration_driven,
+           :deployable?,
+           :service_mesh_integration?,
+           :provider,
+           :proxy_configs, to: :@proxy
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def initialize(proxy, stage = :staging)
+    @proxy = proxy
+    @stage = stage
+  end
+
+  def call
+    case @stage
+    when :staging, :sandbox
+      deploy
+    when :production
+      deploy_production
+    else
+      raise "nope" # ??
+    end
+  end
+
+  def deploy_v2
```

Comment:
Good for now but I would just remove this method later ;) And replace it with the service everywhere

Reply:
This is only because of this line https://github.com/3scale/porta/pull/1470/files#diff-c1ade6e6897661208cbfd3d0c6ebcf8dR43 :/